lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2883,18 +2883,27 @@ void RootEmitterBase::gatherFiles(bool separateModules) {
   for (auto &op : *rootOp.getBody()) {
     auto info = OpFileInfo{&op, replicatedOps.size()};
 
+    SmallString<32> outputPath;
+
+    auto dirAttr = op.getAttrOfType<hw::OutputDirAttr>("output_dir");
+    if (dirAttr) {
+      llvm::errs() << "Found output_dir attribute " << dirAttr << " on " << op
```

Comment:
Use, `op->emitError` instead of `llvm::errs()`

Reply:
Oh, my bad, maybe this `errs()` is intentional. Its also used in other parts of the code. Ignore my comment in that case.

Reply:
I've always been a bit puzzled about the use of `errs` vs. `emitError` in ExportVerilog. I think the `errs` is okay to emit general tooling messages that don't really originate from an operation (like creating a directory failed). But for some things, it might be good to do `op->emitError` because the user will see a message that actually points to the original FIR/Chisel/Edith source location.

Anyway, I would also recommend to either drop the `errs()` for debugging, or add a `#define DEBUG_TYPE export-verilog` at the top of the file, and then do `LLVM_DEBUG(llvm::dbgs() << "Found ..." << "\n")`.

Reply:
Updated this with `LLVM_DEBUG(llvm::dbgs() << ...)`.